tensorflow/compiler/xla/service/gpu/amdgpu_compiler.cc

```diff
@@ -139,436 +135,101 @@ string GetROCDLDir(const HloModuleConfig& config) {
-  // temporary buffers are required to run the computation.
-  TF_ASSIGN_OR_RETURN(
-      std::unique_ptr<BufferAssignment> buffer_assignment,
-      BufferAssigner::Run(
-          module.get(), hlo_schedule->ConsumeHloOrdering(),
-          BufferSizeBytesFunction(),
-          /*color_alignment=*/
-          [](LogicalBuffer::Color) { return kXlaAllocatedBufferAlignBytes; },
-          /*allocate_buffers_for_constants=*/true));
-  DumpHloModuleIfEnabled(*module, *buffer_assignment, "after_optimizations");
-
-  IrEmitterContext ir_emitter_context(module.get(), buffer_assignment.get(),
-                                      stream_exec->platform(), &stream_exec->GetDeviceDescription(),
-                                      &llvm_module);
-
-  HloComputation* entry_computation = module->entry_computation();
-
-  IrEmitterUnnested ir_emitter(module->config(), entry_computation,
-                               &ir_emitter_context);
-
-  TF_RETURN_IF_ERROR(ir_emitter.EmitConstantGlobals());
-
-  {
-    XLA_SCOPED_LOGGING_TIMER("AMDGPUCompiler::RunBackend - IR emission");
-    TF_RETURN_IF_ERROR(entry_computation->Accept(&ir_emitter));
-  }
+namespace {
+absl::optional<bool> CanShareBufferHint(const HloInstruction* user,
+                                        const HloInstruction* operand,
+                                        const ShapeIndex& user_index) {
+  return absl::nullopt;
+}
 
-  if (user_pre_optimization_hook_) {
-    user_pre_optimization_hook_(llvm_module);
-  }
-  string ir_module_string_before_opt;
-  const bool embed_ir_in_executable =
-      module->config().debug_options().xla_embed_ir_in_executable();
-  if (VLOG_IS_ON(3) || embed_ir_in_executable) {
-    ir_module_string_before_opt = llvm_ir::DumpModuleToString(llvm_module);
-    VLOG(3) << "LLVM module before optimizations:";
-    XLA_VLOG_LINES(3, ir_module_string_before_opt);
-  }
+}  // namespace
 
-  const string& ir_dump_directory =
-      module->config().debug_options().xla_dump_to();
+HloDataflowAnalysis::CanShareBuffer AMDGPUCompiler::GetCanShareBuffer() {
+  return &CanShareBufferHint;
+}
 
-   llvm_ir::DumpIrIfEnabled(*module, llvm_module, /*optimized=*/false);
-  {
-    XLA_SCOPED_LOGGING_TIMER(
-        "AMDGPUCompiler::RunBackend - Running LLVM verifier");
-
-    std::string err;
-    llvm::raw_string_ostream err_stream(err);
-
-    // verifyModule() returns true if the module is broken.
-    TF_RET_CHECK(!llvm::verifyModule(llvm_module, &err_stream))
-        << "Invalid LLVM IR before optimizations:\n"
-        << err_stream.str()
-        << "\nThis probably indicates a bug in the HLO -> LLVM IR lowering. "
-           "Rerun with --xla_dump_ir_to to get the IR. ";
-  }
+AMDGPUCompiler::AMDGPUCompiler(se::Platform::Id platform_id)
+    : GpuCompiler(platform_id, amdgpu::kTargetTriple, amdgpu::kDataLayout) {}
 
+GpuVersion AMDGPUCompiler::GetGpuVersion(se::StreamExecutor* stream_exec) {
```

Comment:
I'm seeing a similar function inside of my piece of work as well. Leaving a comment here as a reminder for myself to find a good place to put it into a more generic place.